src/instruction.rs

```diff
@@ -24,6 +25,41 @@ impl StackValue {
             )),
         }
     }
+    pub fn is_same_type(&self, other: &Self) -> bool {
+        match self {
```

Comment:
Could this be shorter by comparing the integer discriminants of the values?

Reply:
ok, I'll try it :)

Reply:
Looks better! changed :point_right: 3f983651c600b604b3b98a4a36173bd2bbc0a270